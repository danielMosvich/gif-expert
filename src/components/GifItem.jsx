import { useEffect, useRef, useState } from "react";

const GifItem = ({ id, title, url, height, original }) => {
  const [showCopy, setShowCopy] = useState(false);
  const alertRef = useRef(null);
  const handleCopyClick = () => {
    navigator.clipboard
      .writeText(original)
      .then(() => {
        setShowCopy(true);
      })
      .catch((err) => {
        console.error("Error al copiar el texto: ", err);
      });
  };
  useEffect(() => {
    if (showCopy) {
      alertRef.current.classList.add("fade");
      alertRef.current.addEventListener("animationend", () => {
        setShowCopy(false);
      });
    }
  }, [showCopy]);
  if (url !== undefined) { 
    return (
      <>
        <div
          style={{
            minWidth: "100%",
            minHeight: height,
            backgroundColor: "#5865f2",
            position: "relative",
          }}
          className="card"
          key={id}
        >
          <img src={url} alt={title} onClick={handleCopyClick} />
          {showCopy && (
            <div ref={alertRef} className="popover">
              Copy in your clipboard
            </div>
          )}
        </div>
      </>
    );
  }
};

export default GifItem;
