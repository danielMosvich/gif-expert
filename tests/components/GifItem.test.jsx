import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe("Component GifItem", () => {
  const id = Math.random().toString(36).substring(7);
  const url = "https://i.imgur.com/xxx.gif" //<- example url
  const original = "https://i.imgur.com/xxx.gif" //<- example url
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(
      <GifItem
        url={url}
        id={id}
        title="Test"
        original={original}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('debe mostrar la imagen con al url', () => {
    render(<GifItem url={url} id={id} title="Test" original={original} />);
    // screen.debug()
    const {src, alt} = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe("Test");
  })
  
});
