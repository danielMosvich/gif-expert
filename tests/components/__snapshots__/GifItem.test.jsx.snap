// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
    style="min-width: 100%; background-color: rgb(88, 101, 242); position: relative;"
  >
    <img
      alt="Test"
      src="https://i.imgur.com/xxx.gif"
    />
  </div>
</div>
`;
